sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -3495,6 +3495,13 @@ class SQLQuerySuite extends QueryTest with SharedSparkSession with AdaptiveSpark
     assert(df4.schema.head.name === "randn(1)")
     checkIfSeedExistsInExplain(df2)
   }
-}
 
+  test("SPARK-31761: test byte, short, integer overflow for (Divide) integral type ") {
```

Comment:
For expression-level changes, I think it's better to add UT as well, to test both codegen and interpreted code path. Can you add a test in `ArithmeticExpressionSuite`?

Reply:
`ArithmeticExpressionSuite` does not go through analyze phase, it evaluates the result. 

So if I add the test case, I still get integer overflow

Reply:
Shall I add another test here to cover codegen also ?

Reply:
ah sorry, as we've switched to the type coercion solution, this test is fine.